server/tests/article.test.js

```diff
@@ -0,0 +1,119 @@
+import 'babel-polyfill'
+import path from 'path'
+import { GraphQLError } from 'graphql'
+import { ApolloServer, gql } from 'apollo-server'
+import { describe, expect, jest, test } from '@jest/globals'
+import { importSchema } from 'graphql-import'
+import { createTestClient } from 'apollo-server-testing'
+import resolvers from '../resolvers'
+import { databaseApi, searchApi } from '../endpoints'
+import { articles } from './mocks'
+
+const typeDefs = importSchema(path.join(__dirname, '../typedefs/index.graphql'))
+const server = new ApolloServer({ typeDefs, resolvers })
+
+const tests = [
+  {
+    name: 'Normal request with a valid search',
+    query: gql`
+            query articles($search: String!) {
+                articles(search: $search) {
+                    id
+                    number
+                    content
+                    keywords
+                    minutesToRead
+                }
+            }
+        `,
+    variables: { search: 'Es obligatorio usar casco con bicicleta?' },
+    response: { data: { articles: ['1'] } },
+    expected: { articles: [articles[0]] }
+  },
+  {
+    name: 'Specific attributes to select',
+    query: gql`
+            query articles($search: String!) {
+                articles(search: $search) {
+                    number
+                    content
+                }
+            }
+        `,
+    variables: { search: 'Es obligatorio usar casco con bicicleta?' },
+    response: { data: { articles: ['1'] } },
+    expected: { articles: [articles[0]].map(({ number, content }) => ({ number, content })) }
```

Comment:
Is this test doing something from within your code. To me it just looks that it's testing the functionality of .map(), something you shouldn't be testing

Reply:
Yees, it's testing to ensure that that only the selected attributes from the request get fetched. So basically, that test case makes sure that GraphQL is fetching the minimum requested data.